src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -86,38 +84,70 @@ const openKeysToItem = (menuItems, selectedPath) => keysToItem(menuItems, select
 }, {});
 
 const CollapsingNavigationMenu = ({ selectedPath = undefined, menuItems, onSelect }) => {
-  const [isNewSelectedPath, setIsNewSelectedPath] = useState(false);
-  const [previousSelectedPath, setPreviousSelectedPath] = useState(selectedPath);
   const [openKeys, setOpenKeys] = useState(openKeysToItem(menuItems[0], selectedPath));
+  const [currentNodeId, setCurrentNodeId] = useState();
```

Comment:
The current node does not need to exist in state as we don't want to trigger a render when simply navigation. Only an expand collapse or selection of an end node should trigger a render. The goal being only the `focus()` should trigger a DOM reflow. There are various ways we can retrieve the currently element without a reflow, inspecting the event's target/currentTarget and using document.activeElement for example.

Reply:
I believe I've resolved this issue, but someone should probably double check my work.